[docs] Document sorting by language (#12033)
yt-dlp/yt-dlp: README.md

 - `hasvid`: Gives priority to formats that have a video stream
 - `hasaud`: Gives priority to formats that have an audio stream
 - `ie_pref`: The format preference

>  - `lang`: The language preference as decided by extractor (usually: indicated by the url > original language > other languages), also (no audio description > with audio description)

Reviewer: I think this is sufficient clarification. Anything beyond this has the potential to be inaccurate
```suggestion
 - `lang`: The language preference as determined by the extractor
```

>  - `lang`: The language preference as decided by extractor (usually: indicated by the url > original language > other languages), also (without audio description > without audio description)

Author: ```suggestion
 - `lang`: The language preference as decided by extractor (usually: indicated by the url > original language > other languages), also (no audio description > with audio description)
```
